sqlmesh/core/engine_adapter/base.py

```diff
@@ -1411,7 +1411,7 @@ def scd_type_2_by_time(
         unique_key: t.Sequence[exp.Expression],
         valid_from_col: exp.Column,
         valid_to_col: exp.Column,
-        execution_time: TimeLike,
+        execution_time: t.Union[TimeLike, exp.Column],
```

Comment:
I guess it's no longer `execution_time` per se. It's more like `valid_to_time` which can sometimes be execution_time